### PR TITLE
Fix handling of ProviderException

### LINKED ISSUE #146: [BUG] AttributeError: 'ProviderError' object has no attribute 'hostname'
Tower has failed WF for which #145 was fixed and merged but now we see the following error with broker:

```
self = <broker.broker.VMBroker object at 0x7fcdca8e0370>
12:10:09  
12:10:09      @mp_decorator
12:10:09      def _checkout(self):
12:10:09          """checkout one or more VMs
12:10:09      
12:10:09          :return: List of Host objects
12:10:09          """
12:10:09          hosts = []
12:10:09          if not self._provider_actions:
12:10:09              raise self.BrokerError("Could not determine an appropriate provider")
12:10:09          for action in self._provider_actions.keys():
12:10:09              provider, method = PROVIDER_ACTIONS[action]
12:10:09              logger.info(f"Using provider ********provider.__name__} to checkout")
12:10:09              try:
12:10:09                  host = self._act(provider, method, checkout=True)
12:10:09                  logger.debug(f"host=********host}")
12:10:09              except exceptions.ProviderError as err:
12:10:09                  host = err
12:10:09              if host:
12:10:09                  hosts.append(host)
12:10:09  >               logger.info(f"********host.__class__.__name__}: ********host.hostname}")
12:10:09  E               AttributeError: 'ProviderError' object has no attribute 'hostname'
12:10:09  
12:10:09  ../../lib64/python3.8/site-packages/broker/broker.py:139: AttributeError
```

Please look into it !

### LINKED ISSUE #172: Properly handle return code after errors
Broker should properly handle RC after hitting errors.
I won't go grepping broker output for `ERROR` - that's purpose of return codes

```broker checkout --workflow deploy-satellite-upgrade --deploy_sat_version 6.11 --deploy_scenario satellite-upgrade --deploy_rhel_version 8 --count 1 --target_cores 6 --target_memory 24GiB
[INFO 221104 15:39:26] Using provider AnsibleTower to checkout
[INFO 221104 15:39:26] Using username and password authentication
[INFO 221104 15:39:28] Waiting for job: 
    API: https://infra-ansible-tower-01/api/v2/workflow_jobs/142076/
    UI: https://infra-ansible-tower-01/#/jobs/workflow/142076/output
[ERROR 221104 15:40:00] ProviderError: AnsibleTower encountered the following error: {'reason': 'Unable to determine failure cause for deploy-satellite-upgrade ar /api/v2/workflow_jobs/142076/'}
[Pipeline] echo
Broker RC for checkout: 0```

Actual Result: 
`RC = 0`

Expected Result:
`RC !=0`

### PR DESCRIPTION
**Pull request by lpramuk**
Fixes #172
Fixes #146

Fixes typos and stylecheck
Updates setup.cfg

(no squash please)

### CODE REVIEW

**Comment by lpramuk** [comment]
I've done nice cleanup indeed but RC remains 0 on ProviderError :sad:

**Comment by lpramuk** [comment]
```
$ broker checkout --workflow blahblah
[INFO 230118 16:23:44] Using provider AnsibleTower to checkout
[INFO 230118 16:23:44] Using token authentication
[ERROR 230118 16:23:46] ProviderError: AnsibleTower encountered the following error: Workflow not found by name: blahblah
$ echo $?
7
```

File: broker/providers/ansible_tower.py [hunk 1]
```diff
@@ -372,7 +372,7 @@ def _compile_host_info(self, host):
                     host_info["_broker_args"]["workflow"] = host.get_related(
                         "last_job"
                     ).summary_fields.source_workflow_job.name
-                except:
+                except Exception:
```
Comment (JacobCallahan) [inline broker/providers/ansible_tower.py:375]: Are these explicit exception classes a flake thing?

Comment (lpramuk) [reply broker/providers/ansible_tower.py:375]: Yes, it is

plain `except:` is actually `except BaseException:` which is wider and includes KeyboardInterupt and SystemExit
so more proper is `except Exception:`

**Review by JacobCallahan — APPROVED** [review]
ACK, thanks for the thorough PR!

**Pull request merged**